docs/05-concepts/10-authentication/05-custom-integration.md

```diff
@@ -0,0 +1,194 @@
+
+### Send token to client
+
+After creating the token, it should be sent to the client. The client is then responsible for storing the token and including it in communication with the server. The token is usually sent in response to a successful sign-in request.
+
+```dart
+class UserEndpoint extends Endpoint {
+  Future<String?> login(
+    Session session,
+    String username,
+    String password,
+  ) async {
+    var identifier = authenticateUser(session, username, password);
+    if (identifier == null) return null;
+
+    var authToken = await session.auth.signInUser(
+      identifier,
+      'username',
+      scopes: {},
+    );
+
+    return '${authToken.id}:${authToken.key}';
+  }
+}
+```
+
+In the above example, the `login` method authenticates the user and creates an auth token. The token is then returned to the client in the format expected by the default token validation.
+
+### Remove auth token
+When the default token validation is used, signing out a user on all devices is made simple with the `signOutUser` method in the `UserAuthentication` class. The method removes all auth tokens associated with the user.
+
+```dart
+class AuthenticatedEndpoint extends Endpoint {
+  @override
+  bool get requireLogin => true;
+  Future<void> logout(Session session) async {
+    await session.auth.signOutUser();
+  }
+}
+```
+
+In the above example, the `logout` endpoint removes all auth tokens associated with the user. The user is then signed out and loses access to any protected endpoints.
+
+#### Remove specific tokens
+The `AuthKey` table stores all auth tokens and can be interacted with in the same way as any other model with a database in Serverpod. To remove specific tokens, the `AuthKey` table can be interacted with directly.
+
+```dart
+await AuthKey.db.deleteWhere(
+  session,
+  where: (t) => t.userId.equals(userId) & t.method.equals('username'),
+);
+```
+
+In the above example, all auth tokens associated with the user `userId` and created with the method `username` are removed from the `AuthKey` table.
+
+
+#### Custom token solution
+If a [custom auth token](#custom-tokens) solution has been implemented, auth token removal must be handled manually. The `signOutUser` method does not provide an interface to interact with other database tables.
+
+## Client setup
+Enabling authentication in the client is as simple as configuring a key manager and placing any token in it. If a key manager is configured, the client will automatically query the manager for a token and include it in communication with the server.
+
+### Configure key manager
+Key managers need to implement the `AuthenticationKeyManager` interface. The key manager is configured when creating the client by passing it as the named parameter `authenticationKeyManager`. If no key manager is configured, the client will not include tokens in requests to the server.
```

Comment:
Mention that for most use cases, you can just use the `FlutterKeyManager` (or whichever we called it).